api/v1alpha1/policyrecommendation_types.go

```diff
@@ -20,28 +20,22 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
 // PolicyRecommendationSpec defines the desired state of PolicyRecommendation
 type PolicyRecommendationSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
-	// Foo is an example field of PolicyRecommendation. Edit policyrecommendation_types.go to remove/update
-
-	WorkloadSpec           WorkloadSpec     `json:"workload"`
-	TargetHPAConfiguration HPAConfiguration `json:"targetHPAConfig"`
-	Policy                 Policy           `json:"policy"`
-	GeneratedAt            metav1.Time      `json:"generatedAt,omitempty"`
-	QueuedForExecution     bool             `json:"queuedForExecution"`
-	QueuedForExecutionAt   metav1.Time      `json:"queuedForExecutionAt,omitempty"`
+	WorkloadMeta            WorkloadMeta     `json:"workload,omitempty"`
+	TargetHPAConfiguration  HPAConfiguration `json:"targetHPAConfig,omitempty"`
+	CurrentHPAConfiguration HPAConfiguration `json:"currentHPAConfig,omitempty"`
+	Policy                  string           `json:"policy,omitempty"`
+	GeneratedAt             *metav1.Time     `json:"generatedAt,omitempty"`
+	TransitionedAt          *metav1.Time     `json:"transitionedAt,omitempty"`
+	QueuedForExecution      *bool            `json:"queuedForExecution,omitempty"`
```

Comment:
What do we gain by converting these fields to pointers and making them non empty? The pointers themselves can be nil.

Reply:
Please refer https://github.com/golang/go/issues/13284 . false values are treated as empty and not updated. This is the suggested solve for it.

Reply:
I didn't want to remove the omitEmpty to ensure that the the controllers/triggers whichever making the changes to policyreco explicitly specify either true/false values to this field and didn't want the empty value to be treated as false.